Bunkum.HttpServer/Endpoints/Middlewares/MainMiddleware.cs

```diff
@@ -262,4 +272,20 @@ public void HandleRequest(ListenerContext context, Lazy<IDatabaseContext> databa
 
         return null;
     }
+
+    private static void TrimToToFirstNullByte(MemoryStream body)
+    {
+        long i = 0;
+        body.Seek(0, SeekOrigin.Begin);
+        int b;
+        while ((b = body.ReadByte()) != -1)
+        {
+            if (b == 0) break;
+
+            i += 1;
+        }
+
+        body.SetLength(i);
```

Comment:
Unsure about the implications of calling this if there are null bytes, maybe check if lengths are equal?

Reply:
I checked the .NET source and it looks like this is probably a safe call. The function we call doesn't explicitly handle this case and will probably a hefty amount of useless operations.

https://source.dot.net/#System.Private.CoreLib/src/libraries/System.Private.CoreLib/src/System/IO/MemoryStream.cs,547